### DISCUSSION TITLE
Best way to property bind an Eloquent model?

### DISCUSSION THREAD

**Discussion by timrspratt** [livewire/livewire#1871] [1. Help]
Hi,

I'm after advice on the best way to do the following, where a property is an eloquent model, which can be changed using a select input dropdown:

```php
class ExampleComponent extends Component
{
    public Organization $organization;

    public function mount()
    {
        return view('livewire.example-component', [
             'organizations' => Organizations::all(),
        ]);
    }
}
```

```blade
<select wire:model="organization">
    @foreach($organizations as $organization)
        <option value="{{ $organization->id }}">{{ $organization->name }}</option>
    @endforeach
</select>
```

 I've been looking at using `organization.id` and adding a validation rule `required|exists:organizations,id` but can't seem to get anything concrete?

Thanks

**Comment by joshhanley** [comment]
I'm not sure if bound models works with select boxes to select a whole model.

I think I would change $organisation to $organisationId and do the rule as you suggested, then you could have a computed property for $organisation.

```php
    public $organizationId;

    public $rules = [ 'organizationId' => 'required|exists:organizations,id'];

    public function getOrganizationProperty()
    {
        return Organisation::find($this->organizationId);
    }
```

Hope this helps!
